apps/etterlatte-vilkaarsvurdering-kafka/src/main/kotlin/vilkaarsvurdering/Vilkaarsvurder.kt

```diff
@@ -0,0 +1,52 @@
+package no.nav.etterlatte.vilkaarsvurdering
+
+import no.nav.etterlatte.libs.common.logging.withLogContext
+import no.nav.etterlatte.libs.common.rapidsandrivers.correlationId
+import no.nav.etterlatte.libs.common.rapidsandrivers.eventName
+import no.nav.etterlatte.rapidsandrivers.EventNames
+import no.nav.etterlatte.rapidsandrivers.EventNames.VILKAARSVURDER
+import no.nav.etterlatte.vilkaarsvurdering.services.VilkaarsvurderingService
+import no.nav.helse.rapids_rivers.JsonMessage
+import no.nav.helse.rapids_rivers.MessageContext
+import no.nav.helse.rapids_rivers.RapidsConnection
+import no.nav.helse.rapids_rivers.River
+import no.nav.helse.rapids_rivers.toUUID
+import org.slf4j.LoggerFactory
+import rapidsandrivers.BEHANDLING_ID_KEY
+import rapidsandrivers.BEHANDLING_VI_OMREGNER_FRA_KEY
+import rapidsandrivers.SAK_ID_KEY
+import rapidsandrivers.behandlingId
+import rapidsandrivers.withFeilhaandtering
+
+internal class Vilkaarsvurder(
+    rapidsConnection: RapidsConnection,
+    private val vilkaarsvurderingService: VilkaarsvurderingService
+) :
+    River.PacketListener {
+    private val logger = LoggerFactory.getLogger(Vilkaarsvurder::class.java)
+
+    init {
+        River(rapidsConnection).apply {
+            eventName(VILKAARSVURDER)
+            validate { it.requireKey(SAK_ID_KEY) }
+            validate { it.requireKey(BEHANDLING_ID_KEY) }
+            validate { it.requireKey(BEHANDLING_VI_OMREGNER_FRA_KEY) }
+            correlationId()
+        }.register(this)
+    }
+
+    override fun onPacket(packet: JsonMessage, context: MessageContext) =
+        withLogContext(packet.correlationId) {
+            withFeilhaandtering(packet, context, VILKAARSVURDER) {
+                val behandlingId = packet.behandlingId
+                val behandlingViOmregnerFra = packet[BEHANDLING_VI_OMREGNER_FRA_KEY].asText().toUUID()
+
+                vilkaarsvurderingService.kopierForrigeVilkaarsvurdering(behandlingId, behandlingViOmregnerFra)
+                packet.eventName = EventNames.BEREGN
+                context.publish(packet.toJson())
+                logger.info(
+                    "Vilkaarsvurdert ferdig for behandling $behandlingId og melding beregningsmelding ble sendt."
+                )
+            }
+        }
```

Comment:
Hadde det vært nice å wrappet dette i en try-catch, og så logge noe spesifikt for hva konsekvensen er for at det kræsjer inne i appen? Slik at det blir lettere å skjønne hva som skjer i loggen når man ser 

"Kunne ikke kopiere forrige vilkårsvurdering fra behandling $forrigeId til omregningsbehandling $nyId på grunn av feil. Dette stopper flyten for alle omregnings/revurderinssaker" 

i stedet for en feil i httpClient get f.eks.

Reply:
withFeilhaandtering gjer try-catch, kanskje godt nok?